frontend/src/metabase/components/TextWidget/TextWidget.unit.spec.tsx

```diff
@@ -10,56 +11,83 @@ const TextInputWithStateWrapper = ({ value }: { value?: number | string }) => {
   );
 };
 
+const PLACEHOLDER_TEXT = "Enter a value...";
+
 describe("TextWidget", () => {
   it("should render correctly", () => {
     render(
       <TextWidget
-        value={"Hello, world!"}
+        value="Hello, world!"
         setValue={jest.fn()}
         focusChanged={jest.fn()}
-      ></TextWidget>,
+      />,
     );
-    expect(screen.getByRole("textbox")).toHaveValue("Hello, world!");
+
+    const textbox = screen.getByPlaceholderText(PLACEHOLDER_TEXT);
+
+    expect(textbox).toHaveValue("Hello, world!");
   });
 
   it("should accept editing", () => {
     render(
-      <TextWidget
-        value={""}
-        setValue={jest.fn()}
-        focusChanged={jest.fn()}
-      ></TextWidget>,
+      <TextWidget value={""} setValue={jest.fn()} focusChanged={jest.fn()} />,
     );
 
-    fireEvent.change(screen.getByRole("textbox"), {
-      target: { value: "Toucan McBird" },
-    });
-    expect(screen.getByRole("textbox")).toHaveValue("Toucan McBird");
+    const textbox = screen.getByPlaceholderText(PLACEHOLDER_TEXT);
+
+    userEvent.paste(textbox, "Toucan McBird");
```

Comment:
TIL `userEvent` has [`paste` function](https://testing-library.com/docs/user-event/clipboard/#paste). That's pretty neat. 🎉